auth/auth.ts

```diff
@@ -43,12 +43,9 @@ export class MockAuth implements firebase.auth.Auth {
     return this.signIn(user, { isNewUser: true });
   }
 
-  fetchProvidersForEmail(email: string): Promise<any> {
-    throw new Error("Method not implemented.");
-  }
-
   fetchSignInMethodsForEmail(email: string): Promise<string[]> {
-    throw new Error("Method not implemented.");
+    const providers = this.store.findByEmail(email)?.providerData || [];
```

Comment:
Having `findByEmail()` return undefined seems like a possible places that we could clean up.
If all other usages I see of `findByEmail()` we are throwing errors, maybe `findByEmail()` should just throw an error itself?
Does `fetchSignInMethodsForEmail()` in Firebase throw when you give it an invalid email?

Either way I am not a big fan of optional chaining directly after a function call.